### PR TITLE
Implement a jump to now button

### PR DESCRIPTION
**Pull request by jadar**

### CODE REVIEW

File: components/EventList.js [hunk 1]
```diff
@@ -15,6 +16,37 @@ export default class EventList extends React.Component {
         await this.props.reload();
     }
 
+    scrollToNow = () => {
+        let currentHour = new Date();
+        currentHour.setMinutes(0, 0, 0);
```
Comment (ghost) [inline components/EventList.js]: Wait, setMinutes takes *3* arguments?? Who knew?

File: components/EventList.js [hunk 2]
```diff
@@ -15,6 +16,37 @@ export default class EventList extends React.Component {
...
+                break;
+            }
+        }
+
+        if (i >= sections.length) {
+            return;
+        }
+
+        this.sectionListRef.scrollToLocation({
+            itemIndex: -1,
```
Comment (ghost) [inline components/EventList.js]: What's the -1 for?

**Comment by ghost** [comment]
I'll try and test this on Android sometime tomorrow, but I'll assume it works as well as it does on iOS

File: components/EventList.js (hunk 2 above)
Comment (blackary) [reply components/EventList.js]: I think there's a bug, and something like this should get the correct item and section indices, and then using both indices to jump to the specific event of the day

```
const getCurrentItem(sections, currentHour) {
    let i = 0;
    let j = 0;
    for (i = 0; i < sections.length; i++) {
        let sectionData = sections[i].data
        for (j = 0; j < sectionData.length; j++) {
            let startDate = new Date(sectionData[j].start);
            if (startDate >= currentHour) {
                return [i, j];
            }
        }
    }
    return [i-1, j-1]
}
```

File: components/EventList.js (hunk 1 above)
Comment (jadar) [reply components/EventList.js]: Yeah, it also sets seconds and ms.

File: components/EventList.js (hunk 2 above)
Comment (jadar) [reply components/EventList.js]: Really what we're doing is wanting to jump to the current section. At least, that's how the previous app had it implemented. I guess the way this app is set up is that it is sectioned by day. The previous app was sectioned by hour. So we should either section the list by hour or implement the above version. For some reason I just assumed last night that it was by hour.

**Review by blackary — APPROVED** [review]

**Review by jadar — COMMENTED** [review]
Hold on, this is because the sections header is really weird. When I was testing with jumping to the first item in a section, it would jump too far because it doesn't account for the height of the headers. We actually need the -1 because it brings the cell to the top with the header.
IMG_2118
IMG_2119

**Pull request merged**